daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2126,9 +2157,11 @@ func (devices *DeviceSet) removeDevice(devname string) error {
 	return err
 }
 
-func (devices *DeviceSet) cancelDeferredRemoval(info *devInfo) error {
+func (devices *DeviceSet) cancelDeferredRemoval(info *devInfo) (bool, error) {
```

Comment:
I understand what you are saying, but devicemapper (cancelDeferredRemoval) coming back with `err==nil` doesn't necessarily means that the deferred removal was cancelled.
It means "Nothing is wrong" but we still need additional information to know if the deferred removal was actually cancelled, and do we need to reschedule it?
 
`err == nil` can mean
a) deferred removal option is not even supported. FALSE
b) the device exists but it is not marked for deferred removal. FALSE
c) the deferred removal was scheduled on the device, and now it is cancelled successfully. TRUE
d) The device was already gone, by the time we reached to cancel the deferred removal. So now it's not needed anymore. FALSE.

Only in case (c) we really need to reschedule the deferred removal, since it was actually cancelled.
I know what you are saying with `os.IsNotExist(err)` however that is a type of error and can be checked. We don't have an error here, and we need more information to make an informed decision.
Unless, you rewrite the entire functionality on how we implement `cancelDeferredRemoval`, I am afraid we would have to go with a boolean variable.